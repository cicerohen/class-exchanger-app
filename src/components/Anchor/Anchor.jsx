import React from 'react';
import PropTypes from 'prop-types';
import Style from './style/Anchor.scss';

const Anchor = ({
  href, onClick, title, children, style,
}) => (
  <a className={style.root} href={href} onClick={onClick} title={title}>
    { children }
  </a>
);

Anchor.defaultProps = {
  href: '#',
  title: undefined,
  onClick: undefined,
  style: Style,
};

Anchor.propTypes = {
  href: PropTypes.string,
  title: PropTypes.string,
  onClick: PropTypes.func,
  style: PropTypes.objectOf(PropTypes.any),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
};

export default Anchor;
