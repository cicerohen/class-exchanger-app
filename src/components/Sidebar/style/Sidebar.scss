@import "./Sidebar.variables";

.root {
  height: 100%;
  background-color: $sidebarBgColor;
}

.container {
  position: relative;
  height: 100%;
}

.sidebarHeaderContainer {
  background-color: $sidebarHeaderBgColor;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-left: $defaultPadding;
  padding-right: $defaultPadding;
  height: 80px;
  > a {
    text-decoration: none;
  }
}

.sidebarBodyContainer {}

.sidebarFooterContainer {
  position: absolute;
  width: 100%;
  bottom: 0;
  background-color: $sidebarFooterBgColor;
}