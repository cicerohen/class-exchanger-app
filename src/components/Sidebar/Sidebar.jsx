import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FaClose } from 'react-icons/lib/fa';
import Anchor from '../Anchor';
import ListItemWithAvatar from '../ListItemWithAvatar';
import Style from './style/Sidebar.scss';

class Sidebar extends Component {
  render() {
    const {
      sidebarTitle, closeSidebarFn, children,
    } = this.props;
    return (
      <aside className={Style.root}>
        <div className={Style.container}>
          <div className={Style.sidebarHeaderContainer}>
            <Anchor>{sidebarTitle}</Anchor>
            <Anchor onClick={closeSidebarFn} title="Close sidebar"><FaClose /></Anchor>
          </div>
          <div className={Style.sidebarBodyContainer}>
            <ListItemWithAvatar
              avatarUrl="https://avatars3.githubusercontent.com/u/819041?s=400&u=e2358cf39d3d95c7603c19f6a23de0c0619e984d&v=4/"
            >
              <div className={Style.user_info}>
                <h3>CÃ­cero Viana</h3>
                <p>cicerohen@gmail.com</p>
              </div>
            </ListItemWithAvatar>
            {children}
          </div>
          <div className={Style.sidebarFooterContainer}>
            {'footer'}
          </div>
        </div>
      </aside>
    );
  }
}

Sidebar.propTypes = {
  sidebarTitle: PropTypes.string.isRequired,
  closeSidebarFn: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
};

export default Sidebar;
