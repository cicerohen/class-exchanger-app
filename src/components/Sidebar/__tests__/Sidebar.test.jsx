import React from 'react';
import { shallow } from 'enzyme';
import { FaClose } from 'react-icons/lib/fa';
import Anchor from '../../Anchor';
import Sidebar from '../Sidebar';

const minimalProps = {
  sidebarTitle: 'Sidebar title',
  closeSidebarFn: jest.fn(),
};

describe('<Sidebar /> Tests', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
  
  it('should render without exploding', () => {
    const wrapper = shallow(
      <Sidebar {...minimalProps}>
        <div>Content</div>
      </Sidebar>
    );
    expect(wrapper).toHaveLength(1);
  });

  it('should render correctly with minimal props', (done) => {
    const wrapper = shallow(<Sidebar {...minimalProps}><div>content</div></Sidebar>);
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should call closeSidebarFn with click on close icon', (done) => {
    const wrapper = shallow(
      <Sidebar {...minimalProps}>
        <div>content</div>
      </Sidebar>
    );
    const anchor = wrapper
    .find(Anchor)
    .filterWhere(n => n.children().type() === FaClose)
    .first()
    .dive()
    .shallow();
    anchor.find('a').first().simulate('click');
    expect(minimalProps.closeSidebarFn.mock.calls.length).toBe(1);
    done();
  });
});