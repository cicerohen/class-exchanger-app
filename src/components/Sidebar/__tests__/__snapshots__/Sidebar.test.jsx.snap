// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Tests should render correctly with minimal props 1`] = `
<aside
  className="root"
>
  <div
    className="container"
  >
    <div
      className="sidebarHeaderContainer"
    >
      <Anchor
        href="#"
        style={null}
      >
        Sidebar title
      </Anchor>
      <Anchor
        href="#"
        onClick={[MockFunction]}
        style={null}
        title="Close sidebar"
      >
        <FaClose />
      </Anchor>
    </div>
    <div
      className="sidebarBodyContainer"
    >
      <ListItemWithAvatar
        avatarUrl="https://avatars3.githubusercontent.com/u/819041?s=400&u=e2358cf39d3d95c7603c19f6a23de0c0619e984d&v=4/"
        style={null}
      >
        <div
          className="user_info"
        >
          <h3>
            CÃ­cero Viana
          </h3>
          <p>
            cicerohen@gmail.com
          </p>
        </div>
      </ListItemWithAvatar>
      <div>
        content
      </div>
    </div>
    <div
      className="sidebarFooterContainer"
    >
      footer
    </div>
  </div>
</aside>
`;
